src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -329,8 +328,8 @@ protected override void ApplyResponseChallenge()
                 return;
             }
 
```

Comment:
We need to implement the Access Denied scenario here as a mirror of Options.LoginPath.